### PR TITLE
ENH: Add nifti12 test that was not run

### PR DESCRIPTION
**Pull request by hjmjohnson**
## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [X] Added test (or behavior not changed)
- [X] Updated API documentation (or API not changed)

### CODE REVIEW

File: Modules/IO/NIFTI/test/itkNiftiImageIOTest12.cxx [hunk 1]
```diff
@@ -25,33 +25,42 @@ itkNiftiImageIOTest12(int ac, char * av[])
...
-  if (ac > 1)
+  if (ac != 3)
   {
-    char * testdir = *++av;
-    --ac;
-    itksys::SystemTools::ChangeDirectory(testdir);
-  }
-  if (ac != 2)
+    std::cerr << "Incorrect command line usage:" << std::endl;
+    std::cerr << av[0] << " <TempOutputDirectory> <filename>" << std::endl;
```
Comment (dzenanz) [inline Modules/IO/NIFTI/test/itkNiftiImageIOTest12.cxx:31]: av[0] -> itkNameOfTestExecutableMacro(av)
from https://github.com/InsightSoftwareConsortium/ITK/blob/091d0f74209a0e1a032e4cb16f5b6b4bab18045e/Modules/Core/TestKernel/include/itkTestingMacros.h#L260-L264

File: Modules/IO/NIFTI/test/itkNiftiImageIOTest12.cxx [hunk 2]
```diff
@@ -69,30 +78,59 @@ itkNiftiImageIOTest12(int ac, char * av[])
...
 
   using Hasher = itk::Testing::HashImageFilter<ImageType>;
-  Hasher::Pointer hasher = Hasher::New();
-  hasher->SetInput(image);
-  hasher->InPlaceOff();
-  hasher->Update();
-
-  std::string originalHash = hasher->GetHash();
 
+  const std::string originalHash = [&]() -> std::string {
```
Comment (dzenanz) [inline Modules/IO/NIFTI/test/itkNiftiImageIOTest12.cxx:82]: Why not name this lambda and reuse it on line 97?

File: Modules/IO/NIFTI/test/itkNiftiImageIOTest12.cxx [hunk 3]
```diff
@@ -69,30 +78,59 @@ itkNiftiImageIOTest12(int ac, char * av[])
...
 
     ITK_TEST_EXPECT_EQUAL(originalHash, readHash);
+
+    ImageType::IndexType threeIndex = { { 3, 3, 3 } };
+    ImageType::PointType origPhysLocationIndexThree;
+    image->TransformIndexToPhysicalPoint<double>(threeIndex, origPhysLocationIndexThree);
+    ImageType::PointType readPhysLocationIndexThree;
+    readImage->TransformIndexToPhysicalPoint<double>(threeIndex, readPhysLocationIndexThree);
+
+    // If the origins, and the spacings, and the direction cosign's are the smae,
```
Comment (dzenanz) [inline Modules/IO/NIFTI/test/itkNiftiImageIOTest12.cxx:114]: `cosign's` -> `cosines`

File: Modules/IO/NIFTI/test/itkNiftiImageIOTest12.cxx (hunk 1 above)
Comment (hjmjohnson) [reply Modules/IO/NIFTI/test/itkNiftiImageIOTest12.cxx:31]: Fixed.  Thank you.

File: Modules/IO/NIFTI/test/itkNiftiImageIOTest12.cxx (hunk 2 above)
Comment (hjmjohnson) [reply Modules/IO/NIFTI/test/itkNiftiImageIOTest12.cxx:82]: Good point.  Done.

**Review by dzenanz — APPROVED** [review]

**Pull request merged**